pkg/build/api/types.go

```diff
@@ -20,6 +20,30 @@ const (
 	DefaultDockerLabelNamespace = "io.openshift."
 )
 
+const (
+	// SecretTypeSource contains data needed to authenticate against SCM servers.
+	//
+	// SecretTypeSource keys are devided into these option groups:
+	//   Basic authentication group. Requires at least one of fields:
+	//     - Secret.Data["username"] - username used to authenticate against SCM server
+	//     - Secret.Data["password"] - password or token needed to authenticate against SCM server
+	//     - Secret.Data["ca.crt"] - CA certificate needed to authenticate against SCM server
+	//   SSH authentication group. Required field:
+	//     - Secret.Data["ssh-privatekey"] - private SSH key needed to authenticate against SCM server
+	SecretTypeSource kapi.SecretType = "Source"
```

Comment:
Putting multiple in the same type doesn't require that the secret can't
have a type that matches well known patterns

On Sep 9, 2015, at 8:45 AM, Cesar Wong <notifications@github.com> wrote:

In pkg/build/api/types.go
<https://github.com/openshift/origin/pull/4433#discussion_r39037058>:

> @@ -20,6 +20,30 @@ const (
>  	DefaultDockerLabelNamespace = "io.openshift."
>  )
>
> +const (
> +	// SecretTypeSource contains data needed to authenticate against SCM servers.
> +	//
> +	// SecretTypeSource keys are devided into these option groups:
> +	//   Basic authentication group. Requires at least one of fields:
> +	//     - Secret.Data["username"] - username used to authenticate against SCM server
> +	//     - Secret.Data["password"] - password or token needed to authenticate against SCM server
> +	//     - Secret.Data["ca.crt"] - CA certificate needed to authenticate against SCM server
> +	//   SSH authentication group. Required field:
> +	//     - Secret.Data["ssh-privatekey"] - private SSH key needed to authenticate against SCM server
> +	SecretTypeSource kapi.SecretType = "Source"

The reason it's a single type is that we're putting everything as keys into
a single secret. The alternative would be to change the BuildSpec to allow
more than one source secret.

—
Reply to this email directly or view it on GitHub
<https://github.com/openshift/origin/pull/4433/files#r39037058>.